.github/set_version.sh

```diff
@@ -1,5 +1,7 @@
+#!/bin/bash
+
 # We set prefix-key to the version from Cargo.toml for Swatinem/rust-cache@v2
 # since the caches seem to accumulate cruft over time;
 # ref https://github.com/PRQL/prql/pull/2407
 
-grep '^version =' Cargo.toml | tr -d ' "' >> $GITHUB_ENV
+echo "version=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[0].version')" >>"$GITHUB_ENV"
```

Comment:
I don't think we should rely on the first package having the version number we want ... we should rather filter for the name to be "prql-compiler".

Reply:
And I think `--raw-output` would be more readable than `-r`.